File: src/main/java/kr/co/bookand/backend/config/JwtSecurityConfig.java
package kr.co.bookand.backend.config;

import kr.co.bookand.backend.config.jwt.exception.ExceptionHandlerFilter;
import kr.co.bookand.backend.config.jwt.JwtFilter;
import kr.co.bookand.backend.config.jwt.TokenFactory;
import lombok.RequiredArgsConstructor;
import org.springframework.security.config.annotation.SecurityConfigurerAdapter;
import org.springframework.security.config.annotation.web.builders.HttpSecurity;
import org.springframework.security.web.DefaultSecurityFilterChain;
import org.springframework.security.web.authentication.UsernamePasswordAuthenticationFilter;

@RequiredArgsConstructor
public class JwtSecurityConfig extends SecurityConfigurerAdapter<DefaultSecurityFilterChain, HttpSecurity> {
    private final TokenFactory tokenFactory;
    private final ExceptionHandlerFilter exceptionHandlerFilter;

    @Override
    public void configure(HttpSecurity builder) {
        JwtFilter customFilter = new JwtFilter(tokenFactory);
        builder.addFilterBefore(customFilter, UsernamePasswordAuthenticationFilter.class);
        builder.addFilterBefore(exceptionHandlerFilter, JwtFilter.class);
    }
}

File: src/main/kotlin/kr/co/bookand/backend/config/JwtSecurityConfig.kt
package kr.co.bookand.backend.config

import kr.co.bookand.backend.config.jwt.JwtFilter
import kr.co.bookand.backend.config.jwt.JwtProvider
import kr.co.bookand.backend.config.jwt.exception.ExceptionHandlerFilter
import org.springframework.security.config.annotation.SecurityConfigurerAdapter
import org.springframework.security.config.annotation.web.builders.HttpSecurity
import org.springframework.security.web.DefaultSecurityFilterChain
import org.springframework.security.web.authentication.UsernamePasswordAuthenticationFilter

class JwtSecurityConfig(
    private val jwtProvider: JwtProvider,
    private val exceptionHandlerFilter: ExceptionHandlerFilter
) : SecurityConfigurerAdapter<DefaultSecurityFilterChain, HttpSecurity>() {
    override fun configure(builder: HttpSecurity) {
        val customFilter = JwtFilter(jwtProvider)
        builder.addFilterBefore(customFilter, UsernamePasswordAuthenticationFilter::class.java)
        builder.addFilterBefore(exceptionHandlerFilter, JwtFilter::class.java)
    }
}